### ISSUE TITLE
Adjustable Timer for Idle Timeout of #1027

### ISSUE THREAD

**Issue by tdoerschel** [nextcloud/text#1715]
Hi,
we use a simple shared text file as a status board (who is in which room right now, who is on vaccation, what is going on etc.),
so everyone read+writes it during the day.

The 30 minute idle timeout is really annoying (see #1027). Also, because it hits often while writing in the text file.

Is there a way to adjust the timeout, or generally increase it to 10 hours?

**Comment by juliushaertl** [comment]
Yes, we should definitely improve this. I' wondering if we could just use the timer for expiring if the window is not in the foreground which should be possible with https://developer.mozilla.org/de/docs/Web/API/Page_Visibility_API

**Comment by kkplein** [comment]
This would be very convenient if it were implemented yes..!

**Comment by max-nextcloud** [comment]
Copying what @mejo- said in #3375:

> Some ideas how to improve idle session handling:
> 
> * Only start idle timer when [page becomes invisible](https://developer.mozilla.org/en-US/docs/Web/API/Page_Visibility_API)
> * Set session to idle after 30 minutes of page being invisible
> * Reset timer once [page becomes visible again](https://developer.mozilla.org/en-US/docs/Web/API/Page_Visibility_API)
> * Automatically reconnect once [page becomes visible](https://developer.mozilla.org/en-US/docs/Web/API/Page_Visibility_API) if session got idle in the meantime
